### PR TITLE
Fix/CodeQL Warnings 2021-09 (Part 1)

### PR DESCRIPTION
**Pull request by stephengtuggy**
Thank you for submitting a pull request and becoming a contributor to the Vega Strike Core Engine.

Please answer the following:

Code Changes:
- [In Progress] Have the PR Validation Tests been run? See https://github.com/vegastrike/Vega-Strike-Engine-Source/wiki/Pull-Request-Validation
- [ ] This is a documentation change only

Issues:
- Please list any related issues

Purpose:
- What is this pull request trying to do? Fix some CodeQL warnings. And in the process, refactor some code so that, for example, the startup sequence on Win32 matches that on other platforms more closely.
- What release is this for? 0.9.x
- Is there a project or milestone we should apply this to? 0.9.x

### CODE REVIEW

File: engine/src/main.cpp [hunk 1]
```diff
@@ -273,9 +273,11 @@ int main( int argc, char *argv[] )
...
     VegaStrikeLogging::VegaStrikeLogger::InitLoggingPart1();
 
-#ifdef WIN32
-    VSFileSystem::InitHomeDirectory();
-#endif
+    // stephengtuggy 2021-09-10 Let's try initializing the directories in the same order
+    // on Windows as on other platforms, and see what happens
+// #ifdef WIN32
+//     VSFileSystem::InitHomeDirectory();
+// #endif
```
Comment (BenjamenMeyer) [inline engine/src/main.cpp:280]: :+1:

File: engine/src/universe_util_generic.cpp [hunk 2]
```diff
@@ -898,43 +907,73 @@ string getSaveInfo( const std::string &filename, bool formatForTextbox )
...
+                ss << "  " << simplePrettySystem( Ships[i] ) << lf;
+            }
         }
     }
     if (!game_options.quick_savegame_summaries) {
         bool hit = false;
         for (set< string >::const_iterator it = campaign_score_vars.begin(); it != campaign_score_vars.end(); ++it) {
             string var = *it;
-            unsigned int curscore = savegame.getMissionData( var ).size()+savegame.getMissionStringData( var ).size();
+            int_fast64_t curscore = savegame.getMissionData( var ).size() + savegame.getMissionStringData( var ).size();
```
Comment (BenjamenMeyer) [inline engine/src/universe_util_generic.cpp:963]: why `int_fast64_t` instead of `int64_t`?

**Review by BenjamenMeyer — COMMENTED** [review]
One question, generally looks good

File: engine/src/main.cpp (hunk 1 above)
Comment (stephengtuggy) [reply engine/src/main.cpp:280]: As a nice bonus, this change seems to have enabled PWCU to start on Windows without crashing.

File: engine/src/universe_util_generic.cpp (hunk 2 above)
Comment (stephengtuggy) [reply engine/src/universe_util_generic.cpp]: OK, I went with `int64_t`.

**Review by BenjamenMeyer — APPROVED** [review]

**Pull request merged**